File: oms-ui-ng2/src/app/common/models/ratecard.ts
import { Product } from './product'
import { AdUnit } from './ad.unit'
import { SeasonalDiscount } from './seasonal.discount'
import { Premium } from './premium'

export class RateCard {

    rateCardId: string;
    createdDate: Date;
    modifiedDate: Date;
    basePrice: number;
    sectionsName: string;
    notes: string;
    isActive: number;
    version: number;
    createdBy: string;
    modifiedBy: string;
    isRatecardrounded: number;
    product: Product;
    adUnits: Array<AdUnit>;
    premiums: Array<Premium>;
    seasonalDiscounts: Array<SeasonalDiscount>;

}

File: oms/oms-media/src/main/java/com/oms/model/RateCard_.java
package com.oms.model;

import java.math.BigDecimal;
import java.util.List;

import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;

@StaticMetamodel(RateCard.class)
public class RateCard_ extends BaseEntity_ {
	public static volatile SingularAttribute<RateCard, BigDecimal> basePrice;
	public static volatile SingularAttribute<RateCard, String> sectionsName;
	public static volatile SingularAttribute<RateCard, String> notes;
	public static volatile SingularAttribute<RateCard, Boolean> isActive;
	public static volatile SingularAttribute<RateCard, Boolean> isRatecardrounded;

	public static volatile SingularAttribute<RateCard, Product> product;
	public static volatile SingularAttribute<RateCard, List<ADUnit>> adUnits;
	public static volatile SingularAttribute<RateCard, List<Premium>> premiums;
//	public static volatile SingularAttribute<RateCard, Product> seasonalDiscounts;

}
